src/main/java/com/lorepo/icplayer/client/module/ordering/OrderingPrintable.java

```diff
@@ -1,93 +1,256 @@
 package com.lorepo.icplayer.client.module.ordering;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.util.HashMap;
 
-import com.google.gwt.user.client.Random;
-import com.lorepo.icf.utils.JavaScriptUtils;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.Style;
+import com.google.gwt.user.client.DOM;
+import com.lorepo.icplayer.client.model.alternativeText.AlternativeTextService;
+import com.lorepo.icplayer.client.printable.Printable.PrintableMode;
 import com.lorepo.icplayer.client.printable.PrintableContentParser;
 import com.lorepo.icplayer.client.printable.PrintableController;
 
 public class OrderingPrintable {
 
-	OrderingModule model = null;
-	PrintableController controller = null;
-	
-	public OrderingPrintable(OrderingModule model, PrintableController controller) {
-		this.model = model;
-		this.controller = controller;
-	}
-	
-	public String getPrintableHTML(String className, boolean showAnswers) {
-		String result = "<div class=\"printable_ic_ordering\" id=\"" + model.getId() +"\">";
-		
-		String[] parsedItems = new String[model.getItemCount()];
-		List<String> unorderedItems = new ArrayList<String>();
-		
-		for (int i = 0; i < model.getItemCount(); i++) {
-			OrderingItem item = model.getItem(i);
-			String parsedItem = "";
-			if (showAnswers) {
-				parsedItem = createItem(item, model.isVertical(), i+1);
-			} else {
-				parsedItem = createItem(item, model.isVertical(), 0);
-			}
-			Integer startingPosition = item.getStartingPosition();
-			if (startingPosition != null) {
-				parsedItems[startingPosition - 1] = parsedItem;
-			} else {
-				unorderedItems.add(parsedItem);
-			}
+    OrderingModule model = null;
+    PrintableController controller = null;
+    OrderingPrintableCollection collection = null;
+    HashMap<String, String> printableState = null;
+
+    public OrderingPrintable(
+            OrderingModule model, PrintableController controller) {
+        this.model = model;
+        this.controller = controller;
+        this.collection = new OrderingPrintableCollection(
+                controller, model.getItemCount());
+    }
+
+    public String getPrintableHTML(String className, boolean showAnswers) {
+        if (this.model.getPrintable() == PrintableMode.NO) {

```

Comment:
Nie można by przenieść tego porównania do modelu OrderingModule?
get isPrintableMode() {
  return this.getPrintable() == !PrintableMode.NO
}
Wyżej jedynie przykładowy pseudokod.